core/iwasm/libraries/lib-wasi-threads/test/stress_test_threads_creation.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+enum CONSTANTS {
+    NUM_ITER = 90000,
+    NUM_RETRY = 5,
+    MAX_NUM_THREADS = 8,
+    RETRY_SLEEP_TIME_US = 4000,
+};
+
+unsigned int threads_executed = 0;
+unsigned int threads_creation_tried = 0;
+
+pthread_mutex_t mutex;
+
+bool is_finished[MAX_NUM_THREADS];
+
+void *
+thread_func(void *arg)
+{
+    __atomic_fetch_add(&threads_executed, 1, __ATOMIC_RELAXED);
+
+    pthread_mutex_lock(&mutex);
+    int num = *(int *)(arg);
+    is_finished[num] = true;
+    pthread_mutex_unlock(&mutex);
+
+    return NULL;
+}
+
+void
+spawn_thread(pthread_t *thread, void *arg)
+{
+    int status_code = -1;
+    unsigned timeout = RETRY_SLEEP_TIME_US;
+    for (int tries = 0; status_code != 0 && tries < NUM_RETRY; ++tries) {
+        status_code = pthread_create(thread, NULL, &thread_func, arg);
+        ++threads_creation_tried;
+        assert(status_code == 0 || status_code == EAGAIN);
+        if (status_code == EAGAIN) {
+            usleep(timeout);
+            timeout *= 2;
+        }
+    }
+
+    if (status_code != 0) {
+        fprintf(stderr,
+                "Threads executed %d. Aborting because of failed retries.",
+                threads_executed);
+    }
+    assert(status_code == 0 && "Thread creation should succeed");
+}
+
+int
+main(int argc, char **argv)
+{
+    pthread_mutex_init(&mutex, NULL);
+    pthread_t threads[MAX_NUM_THREADS];
+    int args[MAX_NUM_THREADS];
+    double percentage = 0.1;
+
+    for (int i = 0; i < MAX_NUM_THREADS; ++i) {
+        is_finished[i] = true;
+        args[i] = i;
+    }
+
+    for (int iter = 0; iter < NUM_ITER; ++iter) {
+        if (iter > NUM_ITER * percentage) {
+            fprintf(stderr, "Spawning stress test is %d%% finished\n",
+                    (unsigned int)(percentage * 100));
+            percentage += 0.1;
+        }
+
+        int thread_num = -1;
+
+        while (thread_num == -1) {
+            pthread_mutex_lock(&mutex);
+            for (int i = 0; i < MAX_NUM_THREADS; ++i) {
+                if (is_finished[i]) {
```

Comment:
do we need to know which thread finished? Wouldn't be enough to know how many threads is running?

Reply:
I want to reuse `pthread_t` array so I need to know which of `threads[i]` I need to rewrite

Reply:
As said in the comment above, you don't need to track it